### PR TITLE
Move 20H2 queues to server 2022

### LINKED ISSUE #64763: system.globalization.tests.icutests.icushouldbeloaded
|Build|Kind|Start Time|
|---|---|---|
[1589182](https://dev.azure.com/dnceng/public/_build/results?buildId=1589182)|[PR 62740](https://github.com/dotnet/runtime/pull/62740)|2022-03-02|
[1589203](https://dev.azure.com/dnceng/public/_build/results?buildId=1589203)|Rolling|2022-03-02|
[1589450](https://dev.azure.com/dnceng/public/_build/results?buildId=1589450)|[PR 64743](https://github.com/dotnet/runtime/pull/64743)|2022-03-02|
[1589641](https://dev.azure.com/dnceng/public/_build/results?buildId=1589641)|[PR 64567](https://github.com/dotnet/runtime/pull/64567)|2022-03-02|
[1590183](https://dev.azure.com/dnceng/public/_build/results?buildId=1590183)|[PR 64330](https://github.com/dotnet/runtime/pull/64330)|2022-03-02|
[1590727](https://dev.azure.com/dnceng/public/_build/results?buildId=1590727)|Rolling|2022-03-02|
[1590775](https://dev.azure.com/dnceng/public/_build/results?buildId=1590775)|[PR 64759](https://github.com/dotnet/runtime/pull/64759)|2022-03-02|
[1590822](https://dev.azure.com/dnceng/public/_build/results?buildId=1590822)|[PR 64766](https://github.com/dotnet/runtime/pull/64766)|2022-03-02|
[1591380](https://dev.azure.com/dnceng/public/_build/results?buildId=1591380)|[PR 64759](https://github.com/dotnet/runtime/pull/64759)|2022-04-02|
[1591573](https://dev.azure.com/dnceng/public/_build/results?buildId=1591573)|[PR 64766](https://github.com/dotnet/runtime/pull/64766)|2022-04-02|
[1591711](https://dev.azure.com/dnceng/public/_build/results?buildId=1591711)|Rolling|2022-04-02|
[1591751](https://dev.azure.com/dnceng/public/_build/results?buildId=1591751)|[PR 64330](https://github.com/dotnet/runtime/pull/64330)|2022-04-02|
[1591746](https://dev.azure.com/dnceng/public/_build/results?buildId=1591746)|[PR 64800](https://github.com/dotnet/runtime/pull/64800)|2022-04-02|
[1592091](https://dev.azure.com/dnceng/public/_build/results?buildId=1592091)|[PR 64734](https://github.com/dotnet/runtime/pull/64734)|2022-04-02|
[1592314](https://dev.azure.com/dnceng/public/_build/results?buildId=1592314)|[PR 64743](https://github.com/dotnet/runtime/pull/64743)|2022-04-02|
[1592168](https://dev.azure.com/dnceng/public/_build/results?buildId=1592168)|[PR 64800](https://github.com/dotnet/runtime/pull/64800)|2022-04-02|
[1596337](https://dev.azure.com/dnceng/public/_build/results?buildId=1596337)|Rolling|2022-07-02|

### PR DESCRIPTION
**Pull request by safern**
Fixes: #64763 

The 20H2 queues are Server Core, and Server Core doesn't contain icu.dll on the SysWOW64 directory, so the tests that expect ICU to be loaded on SKUs newer or equal to 1903 fail when running on x86.

Since we already have coverage for Server Core on the docker images, move to the next Data Center LTS release which is Server 2022.

FYI: @tarekgh

### CODE REVIEW

**Review by hoyosjs — APPROVED** [review]

**Review by sbomer — APPROVED** [review]
Thanks! Does `IsIcuCompatiblePlatform` need to be updated?

https://github.com/dotnet/runtime/blob/59e643b9932122a12e71ceb360a9ae3bb9420dec/src/libraries/System.Globalization/tests/IcuTests.cs#L12-L13

**Comment by safern** [comment]
> Thanks! Does IsIcuCompatiblePlatform need to be updated?

I'm waiting for confirmation from the Windows team that this is indeed expected (the missing files on wow64), if they confirm that, we need to update that condition to detect if we are in server core and x86.

**Comment by safern** [comment]
Libraries test run on windows are now green. Will merge to unblock other PRs.

**Pull request merged**
